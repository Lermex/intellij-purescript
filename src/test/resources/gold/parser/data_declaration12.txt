Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSObjectTypeImpl(ObjectType)
            PsiElement({)('{')
            PSRowImpl(Row)
              <empty list>
            PsiElement(})('}')
            PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PsiElement(()('(')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSObjectTypeImpl(ObjectType)
                PsiElement({)('{')
                PSRowImpl(Row)
                  <empty list>
                PsiElement(})('}')
          PsiElement())(')')