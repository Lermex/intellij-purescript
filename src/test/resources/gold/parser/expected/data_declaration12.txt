Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PSASTWrapperElement(Row)
              <empty list>
            PsiElement(})('}')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('Foo')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(ObjectType)
                PsiElement({)('{')
                PSASTWrapperElement(Row)
                  <empty list>
                PsiElement(})('}')
          PsiElement())(')')